gluonnlp/model/highway.py

```diff
@@ -0,0 +1,108 @@
+# coding: utf-8
+
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+"""Highway layer."""
+
+from __future__ import absolute_import
+from __future__ import print_function
+
+__all__ = ['Highway']
+
+from mxnet import gluon
+from mxnet.gluon import nn
+
+
+class Highway(gluon.Block):
+    r"""Highway network.
+
+    We implemented the highway network proposed in the following work::
+
+        @article{srivastava2015highway,
+          title={Highway networks},
+          author={Srivastava, Rupesh Kumar and Greff, Klaus and Schmidhuber, J{\"u}rgen},
+          journal={arXiv preprint arXiv:1505.00387},
+          year={2015}
+        }
+
+    A Highway layer is defined as below:
+
+    .. math::
+        y = (1 - t) * x + t * f(A(x))
+
+    which is a gated combination of a linear transform and a non-linear transform of its input,
+    where :math:`x` is the input tensor, :math:`A` is a linear transformer,
+    :math:`f` is an element-wise non-linear transformer,
+    and :math:`t` is an element-wise transform gate, and :math:`t` refers to carry gate.
+
+    Parameters
+    ----------
+    input_size : int
+        The dimension of the input tensor.  We assume the input has shape ``(batch_size,
+        input_size)``.
+    num_layers : int
+        The number of highway layers to apply to the input.
+    activation : nn.Activation
+        The non-linear activation function.
+    """
+
+    def __init__(self,
+                 input_size,
+                 num_layers,
+                 activation=nn.Activation('relu'),
+                 **kwargs):
+        super(Highway, self).__init__(**kwargs)
+        self._input_size = input_size
+        self._num_layers = num_layers
+
+        with self.name_scope():
+            self.hnet = nn.HybridSequential()
+            with self.hnet.name_scope():
+                for _ in range(self._num_layers):
+                    self.hnet.add(nn.Dense(units=self._input_size * 2, in_units=self._input_size,
+                                           use_bias=True, flatten=False))
+            self._activation = activation
+
+    def set_bias(self):
+        for _, layer in enumerate(self.hnet):
+            layer.bias.data()[self._input_size:] = 1
```

Comment:
what does this do and why is it a public method?

Reply:
consider using a bias_initializer in constructor for the same effect.

Reply:
Should this be `-1` (negative one)? Ie, does this initialize the bias of the transform gate? If so, it seems the following from the NIPS paper applies: 

> We use the transform gate defined as T(x) = σ(W_T^T x + b_T ) , where W_T is the weight matrix and b_T the bias vector for the transform gates.  This suggests a simple initialization scheme which is independent of the nature of H : b T can be initialized with a negative value (e.g. -1, -3 etc.) such that the network is initially biased towards carry behavior.

Reply:
@szha based on offline discussion, we are using a highwaybias initializer now. Thanks 
@leezu Yes, after refering back to the paper, we are setting up the bias of the transform gate as -2 according to the AAAI paper. Thanks